apps/model-ad/api-next/src/main/java/org/sagebionetworks/model/ad/api/next/util/ApiHelper.java

```
@@ -15,24 +16,96 @@ public final class ApiHelper {
+
+  /**
+   * Estimated average size per extra part in cache key construction.
+   * Used to pre-allocate StringBuilder capacity.
+   */
+  private static final int ESTIMATED_PART_SIZE = 20;
+
   private ApiHelper() {}
 
-  public static HttpHeaders createNoCacheHeaders(MediaType mediaType) {
-    HttpHeaders headers = new HttpHeaders();
-    headers.setCacheControl(CACHE_CONTROL_VALUE);
-    headers.setPragma("no-cache");
-    headers.setExpires(0);
-    headers.setContentType(mediaType);
-    return headers;
+  /**
+   * Validates that sortFields and sortOrders have matching element counts.
+   * Note: Presence and non-emptiness are enforced by OpenAPI schema validation
+   * (@NotNull @Size(min = 1)), so this method only validates array length matching.
+   *
+   * @param sortFields List of sort field names (required, non-empty per OpenAPI schema)
+   * @param sortOrders List of sort orders (required, non-empty per OpenAPI schema)
+   * @throws IllegalArgumentException if arrays have different lengths
+   */
+  public static void validateSortParameters(List<String> sortFields, List<?> sortOrders) {
+    int fieldCount = sortFields.size();
+    int orderCount = sortOrders.size();
+
+    if (fieldCount != orderCount) {
+      throw new IllegalArgumentException(
+        String.format(
+          "sortFields and sortOrders must have the same number of elements. " +
+          "Got %d field(s) and %d order(s)",
+          fieldCount,
+          orderCount
+        )
+      );
+    }
+  }
+
+  /**
+   * Creates a Spring Data Sort object from sortFields and sortOrders lists.
+   * Returns Sort.unsorted() if no sort parameters are provided.
+   *
+   * @param sortFields list of field names to sort by
+   * @param sortOrders list of sort orders (1 for ascending, -1 for descending)
+   * @return Sort object for use with Spring Data repositories
+   */
+  public static Sort createSort(
+    @Nullable List<String> sortFields,
+    @Nullable List<Integer> sortOrders
+  ) {
```

Could we call `validateSorParameters` within `createSort`, so we can't forget to call validate before calling create? That would also allow us to remove the validate call in each CT's DelegateImpl, but still fail fast in the CT's Service since `createSort` is called before database operations